src/wallet/test/wallet_tests.cpp

```diff
@@ -952,29 +952,14 @@ BOOST_FIXTURE_TEST_CASE(wallet_sync_tx_invalid_state_test, TestingSetup)
 
     // Now the bad case:
     // 1) Make db always fail
-    // 2) Add transaction that spends the previously created transaction
-    // 3) Check what happen now that the db failed to write the tx. Verifying
-    //    the available balance of this new tx and the old one (which, as the
-    //    output is spent, should at least, be marked dirty)
-
-    dynamic_cast<FailDatabase&>(wallet.GetDatabase()).m_pass = false;
+    // 2) Try to add a transaction that spends the previously created transaction and
+    //    verify that we are not moving forward if the wallet cannot store it
+    static_cast<FailDatabase&>(wallet.GetDatabase()).m_pass = false;
```

Comment:
In 4bd0d0140e2a27e604ca0febcd485f468705c0ed "wallet: guard and alert about a wallet invalid state during chain sync"

`static_cast` change should be in the previous commit.